### ISSUE TITLE
clang-tidy check misc-definitions-in-headers offers no fix as described in the documentation

### ISSUE THREAD

**Issue by sean-mcmanus** [llvm/llvm-project#55093]
I'm using clang 14.0.0. Docs at https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/list.html show check misc-definitions-in-headers as having a fix available, but I don't get any fix and the docs at https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/misc-definitions-in-headers.html don't give info on how to enable a fix.

Repro is with
```cpp

#include <test.h>

int main()
{
    return 0;
}
```
with test.h
```cpp
int i;
```

I'm using --export-fixes (I assume that doesn't matter).

**Comment by njames93** [comment]
`--export-fixes` should produce a yaml file that contains the fixes, what does the output of clang-tidy say?

**Comment by sean-mcmanus** [comment]
Yes, the fix is not in the yaml file, like they are with all the other checks.

**Comment by sean-mcmanus** [comment]
I've added a screenshot of the .yml file:

![image](https://user-images.githubusercontent.com/19859882/165374905-3250a1ad-6a8b-4e8c-b653-aa2ac713cd86.png)

Command line is about as simple as I can make it so I don't see why it would be reproing for others:
"c:\path/clang-tidy.exe" --checks=misc-definitions-in-headers --header-filter=.* --export-fixes="C:\path\fixes.yaml" "C:/path/test.cpp" -- -std=c++17 -xc++ -I"C:/path/test"

**Comment by njames93** [comment]
Ah, the only fix-it the check currently provides is to turn a function defined in a header into an inline function.
Though there is merit to make the check also turn variables into inline variables if compiling for c++17 or newer.

**Comment by sean-mcmanus** [comment]
Great -- that helped -- I got the fix working for functions. I didn't realize there was a difference. If a note could at least be added to the docs that would help.
